src/gen7_mfd.c

```diff
@@ -1853,37 +1853,30 @@ gen7_mfd_jpeg_decode_init(VADriverContextP ctx,
         int v2 = pic_param->components[1].v_sampling_factor;
         int v3 = pic_param->components[2].v_sampling_factor;
 
-        if (h1 == 2 && h2 == 1 && h3 == 1 &&
-            v1 == 2 && v2 == 1 && v3 == 1) {
+        if (h1 == 2 * h2 && h2 == h3 &&
+            v1 == 2 * v2 && v2 == v3) {
             subsampling = SUBSAMPLE_YUV420;
             fourcc = VA_FOURCC_IMC3;
-        } else if (h1 == 2 && h2 == 1 && h3 == 1 &&
-                   v1 == 1 && v2 == 1 && v3 == 1) {
+        } else if (h1 == 2 * h2  && h2 == h3 &&
+                   v1 == v2 && v2 == v3) {
             subsampling = SUBSAMPLE_YUV422H;
             fourcc = VA_FOURCC_422H;
-        } else if (h1 == 1 && h2 == 1 && h3 == 1 &&
-                   v1 == 1 && v2 == 1 && v3 == 1) {
+        } else if (h1 == h2 && h2 == h3 &&
+                   v1 == v2  && v2== v3) {
             subsampling = SUBSAMPLE_YUV444;
             fourcc = VA_FOURCC_444P;
-        } else if (h1 == 4 && h2 == 1 && h3 == 1 &&
-                   v1 == 1 && v2 == 1 && v3 == 1) {
+        } else if (h1 == 4 * h2 && h2 ==  h3 &&
+                   v1 == v2 && v2 == v3) {
             subsampling = SUBSAMPLE_YUV411;
             fourcc = VA_FOURCC_411P;
-        } else if (h1 == 1 && h2 == 1 && h3 == 1 &&
-                   v1 == 2 && v2 == 1 && v3 == 1) {
-            subsampling = SUBSAMPLE_YUV422V;
-            fourcc = VA_FOURCC_422V;
-        } else if (h1 == 2 && h2 == 1 && h3 == 1 &&
-                   v1 == 2 && v2 == 2 && v3 == 2) {
-            subsampling = SUBSAMPLE_YUV422H;
-            fourcc = VA_FOURCC_422H;
-        } else if (h1 == 2 && h2 == 2 && h3 == 2 &&
-                   v1 == 2 && v2 == 1 && v3 == 1) {
+        } else if (h1 == h2 && h2 == h3 &&
+                   v1 == 2 * v2 && v2 == v3) {
             subsampling = SUBSAMPLE_YUV422V;
             fourcc = VA_FOURCC_422V;
         } else
             assert(0);
-    } else {
+    }
+    else {
         assert(0);
```

Comment:
Please rebase your branch against github.com:01org/intel-vaapi-driver.git, and run ./autogen.sh to install the pre-commit hook which will check your coding style. then commit your change again.